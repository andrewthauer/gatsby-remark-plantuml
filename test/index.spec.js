const testRemarkPlugin = require(`./test-remark-plugin`)

describe(`remark plantuml plugin`, () => {
  beforeEach(() => {
    jest.resetModules()
  })

  describe(`ignores codeblocks that are not plantuml`, () => {
    it(`with lang=''`, async () => {
      const code = `\`\`\`\n// Fake\n\`\`\``
      await testRemarkPlugin.testPlugin({
        code,
      })
    })

    it(`with lang='js'`, async () => {
      const code = `\`\`\`js\n// Fake\n\`\`\``

      await testRemarkPlugin.testPlugin({
        code,
      })
    })
  })
})
