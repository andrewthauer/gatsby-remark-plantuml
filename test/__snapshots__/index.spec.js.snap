// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remark plantuml plugin ignores codeblocks that are not plantuml with lang='' 1`] = `
Object {
  "children": Array [
    Object {
      "lang": null,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 15,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "code",
      "value": "// Fake",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 15,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark plantuml plugin ignores codeblocks that are not plantuml with lang='js' 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js",
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 17,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "code",
      "value": "// Fake",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 17,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
